lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -655,27 +658,30 @@ class ScaledBlockedToMMAv5
       else if (dotOp.getBElemType() == ScaleDotElemType::E2M1)
         IsBMixedPrecFp4 = true;
     }
-
+    bool isMMAv5Fp4PaddedLhs = IsAMixedPrecFp4 || !dotOp.getLhsKPack();
+    bool isMMAv5Fp4PaddedRhs = IsBMixedPrecFp4 || !dotOp.getRhsKPack();
```

Comment:
I don't understand this change. Why do we want to pad the SMEM if `isMixedPrecFp4` is false?

Reply:
We do want to padd if we need to transpose right? As we need to use `mxf8f6f4`  message in this case. Or would pure mxfp4 not need padding in this case?

Reply:
added a comment

Reply:
also added case for pure mxfp4 in test_matmul

Reply:
Interesting, I didn't think about using mxf864 for `mxfp4 x mxfp4` case. Glad to see more uses of the padded layout!

Reply:
funny, I thought we had discussed it some time back and it was your idea, anyway :)